src/util.js

```diff
@@ -107,3 +107,51 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+// TODO: handle different kinds of errors
+function handleVarlinkCallError(ex) {
+    console.error("Failed to do varlinkcall:", ex, JSON.stringify(ex));
+}
+
+export function updateContainers() {
+    let newContainers = {};
+    let newContainersStats = {};
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                .then(reply => {
+                    let newContainersMeta = reply.containers;
+                    let inspectRet = newContainersMeta.map((container) => {
+                        return varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id});
+                    });
+                    Promise.all(inspectRet)
+                            .then((replies) => {
+                                replies.map((reply) => {
+                                    let ctrInspectRet = JSON.parse(reply.container);
+                                    newContainers[ctrInspectRet.ID] = ctrInspectRet;
+                                });
+                            })
+                            .catch(ex => {
+                                handleVarlinkCallError(ex);
+                                reject(ex);
+                            });
+
+                    let containerStatsRet = newContainersMeta.filter(ele => ele.status === "running")
+                            .map((container) => {
+                                return varlinkCall(PODMAN, "io.podman.GetContainerStats", {name: container.id});
+                            });
+                    Promise.all(containerStatsRet)
+                            .then((replies) => {
+                                replies.map((reply) => {
+                                    let ctrStatsRet = reply.container;
+                                    newContainersStats[ctrStatsRet.id] = ctrStatsRet;
+                                });
+                                resolve({newContainers: newContainers, newContainersStats: newContainersStats});
```

Comment:
I just noticed that this is a race condition. It could be that all the `GetContainerStats()` calls finish before all the `InspectContainer()` calls -- in that case the promise would be resolved too early, with `newContainers` still being empty or incomplete. So instead, below this you need another

    Promise.all(inspectRet.concat(containerStatsRet)).then(() => resolve(..));

You don't need to handle errors there, these are already handled here.